### ISSUE TITLE
Integrate `krel obs` into the release process

### ISSUE THREAD

**Issue by saschagrunert** [kubernetes-sigs/cri-tools#1248]
First of all we have to publish the release artifacts to the Google Cloud Bucket:

```
kpromo gh --org kubernetes-sigs --repo cri-tools --bucket k8s-artifacts-cri-tools --release-dir release --tags v1.28.0
```
Publishes the GitHub artifacts in https://console.cloud.google.com/storage/browser/k8s-artifacts-cri-tools/release

Then we need to run the following commands to release cri-tools in OBS: 

```
krel obs stage --packages cri-tools --version 1.28.0 --project isv:kubernetes:core:stable:v1.28:build --nomock
```
Runs a Google Cloud Build job and stages:
https://build.opensuse.org/package/show/isv:kubernetes:core:stable:v1.28:build/cri-tools

```
krel obs release --packages cri-tools --project isv:kubernetes:core:stable:v1.28:build --nomock
```
Runs a Google Cloud Build job and releases:
https://build.opensuse.org/package/show/isv:kubernetes:core:stable:v1.28/cri-tools.20230814075509

Goal is to integrate the steps above into our release workflow, containing the right amount of permissions:
https://github.com/kubernetes-sigs/cri-tools/blob/master/.github/workflows/release.yml

@kubernetes-sigs/release-engineering we need:

- OBS credentials to run `krel obs`
- A service account token to access  `k8s-artifacts-cri-tools`

**Comment by xmudrii** [comment]
I can help with this. I'll add this issue to our project board so that we properly follow it.

**Comment by afbjorklund** [comment]
Shouldn't this be in kubernetes-release, rather than cri-tools ?

**Comment by saschagrunert** [comment]
The automation will happen in this repository, we can also consider moving the specfile here.

**Comment by saschagrunert** [comment]
@kubernetes-sigs/release-engineering how about:

1. Creating a bot account similar to CRI-O to be able to publish packages for cri-tools
2. Move the specfile into this repository
3. Create an publish them similar to https://github.com/cri-o/packaging, but triggered by tags as part of this GitHub action: https://github.com/kubernetes-sigs/cri-tools/blob/master/.github/workflows/release.yml

**Comment by xmudrii** [comment]
@saschagrunert I think this is a solid plan. I would only try to avoid having a new repo for packaging (because getting repos in k or k-sigs is a more complicated process and it's not scalable).

**Comment by xmudrii** [comment]
@saschagrunert Yes, please create an issue in k/k8s.io to track getting a new service account.

**Comment by saschagrunert** [comment]
@xmudrii closing the loop: https://github.com/kubernetes/k8s.io/issues/5957

**Comment by saschagrunert** [comment]
We still need this, and could also think about proposing an automated vendor PR for the RC of a new Kubernetes version. This could include a hint that we may want to cut a release afterwards.

**Comment by saschagrunert** [comment]
@xmudrii do you plan to work in this before v1.31 get's released?

**Comment by xmudrii** [comment]
@saschagrunert Yes, I'll be looking into this issue this week.
/assign

**Comment by saschagrunert** [comment]
Let me know if you need anything, would be awesome to deliver it this cycle.

**Comment by saschagrunert** [comment]
@xmudrii I'm planning to release cri-tools v1.31 this week. Do we still want to invest in automation or plan that for 1.32?

**Comment by xmudrii** [comment]
It's on my TODO list and we'll have it in place for v1.32. :shipit: Sorry for delay!

**Comment by afbjorklund** [comment]
> It's on my TODO list and we'll have it in place for v1.32. 

This seems obsolete, now that that packages are no longer required to be published in Kubernetes repository?

* https://github.com/kubernetes/release/issues/4156

The suggestion was to remove it, already for v1.35.

**Comment by saschagrunert** [comment]
Sounds good to me, let's close this one out.

**Comment by xmudrii** [comment]
Following up on https://github.com/kubernetes/release/issues/4156#issuecomment-3396904039

I think we can consider removing `cri-tools` as a dependency from other packages. This should be possible because of https://github.com/kubernetes/kubeadm/issues/3064

However, I'm not sure that we should stop building the `cri-tools` package. First of all, the `cri-tools` project is a subproject of Kubernetes SIG Node and as a such can build and host packages on the community owned infrastructure. That's not the case for `kubernetes-cni`, which is an external project that we repackage.

Secondly, I think `cri-tools` is something that is widely used out there, it's sometimes the only way to get something done with containers on a node (e.g. debugging, fetching logs, etc). 

Given all that, I think it makes sense to consider continuing providing package builds for `cri-tools`.

@saschagrunert Would be nice to get some input from you on this

**Comment by afbjorklund** [comment]
The main difference was that the Kubernetes binaries were the same as the ones on dl.k8s.io, while the `crictl` was downloaded from the GitHub project (even if "2nd party")

The documentation has not been updated, and is pointing to the wrong version in several places

So it seemed better to have users do their own re-packaging, like they have to do for the runtime and plugins?

**Comment by xmudrii** [comment]
I would say that's not really a problem here. Even if those binaries are coming from GitHub and not `dl.k8s.io`, those binaries are produced by the Kubernetes project and hence can be packaged and served from the community infrastructure.

I'm not sure I understand what's the problem with documentation, if you can point where the issue is exactly, someone can have a look.

**Comment by afbjorklund** [comment]
I removed the homepage link to the Kubic repo...

* https://github.com/kubernetes-sigs/cri-tools/issues/1806

The docs have not been updated in a while (1.31).

* https://github.com/kubernetes/website/issues/35921

The source for 1.34 is pointing to 1.33

https://github.com/kubernetes-sigs/cri-tools/blob/v1.34.0/README.md

```
VERSION="v1.33.0"
wget https://github.com/kubernetes-sigs/cri-tools/releases/download/$VERSION/crictl-$VERSION-linux-amd64.tar.gz
sudo tar zxvf crictl-$VERSION-linux-amd64.tar.gz -C /usr/local/bin
rm -f crictl-$VERSION-linux-amd64.tar.gz
```

The package for 1.34 is pointing to 1.32

https://github.com/kubernetes/release/blob/master/cmd/krel/templates/latest/cri-tools/README.md

```
VERSION="v1.32.0"
wget https://github.com/kubernetes-sigs/cri-tools/releases/download/$VERSION/crictl-$VERSION-linux-amd64.tar.gz
sudo tar zxvf crictl-$VERSION-linux-amd64.tar.gz -C /usr/local/bin
rm -f crictl-$VERSION-linux-amd64.tar.gz
```

Seems to be about building vs tagging order?

**Comment by afbjorklund** [comment]
But I'm OK with keeping one (or both) of the packages, as long they are built and updated with the release...

The dependencies should probably be removed, either way? Then users can decide, if they also want to install /usr/bin/crictl and
/opt/cni/bin (or not)

**Comment by saschagrunert** [comment]
Yeah I'm also a bit on the fence here. On one side it would be good to have less to maintain, especially since cri-tools is owned by SIG Node. OTOH is an integral debugging tool within a cluster, so I'm preferring to keep it for now.

**Comment by afbjorklund** [comment]
I looked at the packages for the RC, and they are still wrong. So I think we have to suffer this for another release?

Package: kubeadm
Version: **1.35.0~rc.1-1.1**
Priority: optional
Section: admin
Maintainer: Kubernetes Authors <dev@kubernetes.io>
Installed-Size: 72.4 MB
Depends: **cri-tools (>= 1.30.0)**
Homepage: https://kubernetes.io

Package: kubelet
Version: **1.35.0~rc.1-1.1**
Priority: optional
Section: net
Maintainer: Kubernetes Authors <dev@kubernetes.io>
Installed-Size: 58.2 MB
Depends: iptables (>= 1.4.21),**kubernetes-cni (>= 1.2.0)**,mount,util-linux,libc6
Homepage: https://kubernetes.io

It is fine if the packages remain, but they should not be added as dependencies (and thus be installed by default)

`cri-tools/unknown,now 1.34.0-1.1 amd64 [installed]`

`kubernetes-cni/unknown,now 1.7.1-1.1 amd64 [installed]`

Note that it is still using the older versions (for CRI and CNI).

But at least the packages are not _missing_, always something?
